### DISCUSSION TITLE
Announcing Snippets and Traffic Splitting for Azure Static Web Apps

### DISCUSSION THREAD

**Discussion by thomasgauvin** [Azure/static-web-apps#1310] [Announcements]
Hi everyone,

We’re excited to announce two new features, [Snippets](https://aka.ms/swa/snippets) and [Traffic Splitting](https://aka.ms/swa/traffic-splitting), to improve your development workflows and improve development productivity! 🎉

## Snippets:

With snippets for Azure Static Web Apps, you can specify code that will be injected into each page of your website without having to change your application’s source code. Snippets are configured on the Static Web Apps resource and get injected at runtime by the Static Web Apps host (when the site is being served).
 
When providing your snippet, you can indicate which environment it should be applied to, and you can configure the code snippet to be appended or prepended to the head or the body or your website’s pages. 

Snippets can be used to add analytics scripts to each page of your website, enabling you to track website performance and engagement. They can also be used to add global UI elements which are shared across multiple pages, such as navigation menus, banners or shared JavaScript code for dynamic behavior such as chat widgets or pop-ups.
 
![image](https://github.com/Azure/static-web-apps/assets/35609369/3a3fc090-a413-4eb8-9b10-dcdd5db86587)

 
## Traffic splitting:

Traffic Splitting enables you to share traffic to your Azure Static Web Apps across and specify the amount of traffic share each environment receives. This feature allows you to gradually roll out new versions of your site and test different versions with real users.
 
For example, you can use traffic splitting to gradually roll out changes to your site without affecting all users at once, which can reduce the risk of errors or negative feedback and allow you to adjust quickly. Traffic splitting can also be used to implement A/B testing, whereby separate versions of your website are served across your users and analytics are used to determine the impact on performance of various changes.
 
 
![image](https://github.com/Azure/static-web-apps/assets/35609369/3e4cf027-0bca-4bae-be9e-91c5f951e67f)

## Try it out & let us know what you think!

You can access Snippets from the Configuration tab of your Static Web Apps resource in the Azure Portal, and you can access Traffic Splitting from the Environments tab. We’re constantly listening for feedback so let us know what you think!

Check out our docs to get started with [Snippets](https://aka.ms/swa/snippets) and [Traffic Splitting](https://aka.ms/swa/traffic-splitting) 🚀

**Comment by chris-dnv** [comment]
Hello,   I was hoping to add the code found here to get more analytics for my static web app [https://learn.microsoft.com/en-us/azure/azure-monitor/app/javascript-sdk?tabs=javascriptwebsdkloaderscript](url) however the code is over the 2048 characters limit and I cannot add the snippet.

**Reply by thomasgauvin** [reply]
We're tracking this on our backlog to double the current characters limit, thanks for the feedback @chris-dnv

**Comment by ChristopherHaws** [comment]
@thomasgauvin Is there a way to automate the deployment of snippets via bicep/arm or az cli?

Thanks!

**Reply by gavar** [reply]
We would also love to see a way to provide snippets via bicep.

**Reply by thomasgauvin** [reply]
Logging this to our backlog, thanks!

**Reply by danielrobinson95** [reply]
@thomasgauvin, any update on this?

**Reply by chris-kruining** [reply]
@thomasgauvin do you have a link to the backlog item perhaps? I couldn't find it in the Azure/bicep repo

**Reply by elmalakai** [reply]
I'm surprised the latest schema for SWA still doesn't have snippets in it. Is there any way we an automate this? We rely on snippets heavily. We even control which API's endpoints we use via this because the linked API backend was too flaky and had it's own set of timeouts, closing connection in odd ways.

**Comment by ppi-kosipenko** [comment]
Also need a way to configure Traffic splitting via Az PowerShell and/or CLI, so it can be automated through the pipeline.

**Reply by danielrobinson95** [reply]
Does this exist yet?

**Reply by VladimirUsov** [reply]
2 years after, this great feature is still in preview and still missing CLI, why?

**Comment by edgariscoding** [comment]
I would love to know what goes on at Microsoft when announcing new features/functionalities. It seems like they release things as one offs and then just forget about them completely. How can something be in preview for two years with zero updates? It was abandoned before it was even released.
